# 단일표본 t검정

- 표본 데이터 로딩
- 표본 데이터가 정규성 따르는지 검정하기
- 단일표본 t검정 \
    - p-value < 0.05 : 귀무가설 기각, 대립가설 채택, 대립가설이 통계적으로 유의미 해석 가능
    - p-value >= 0.05 : 귀무가설 채택, 대립가설 기각

## 정규성 검정하기

In [4]:
float('150.27')

150.27

In [8]:
# data 불러오기
with open("./datas2/성인여성_키_데이터.txt", "r") as f:
    data = f.read().split('\n')
    #print(list(map(float, data)))
    data = list(map(float, data))
    print(data) 

[150.27, 142.94, 160.99, 157.48, 153.46, 137.5, 154.94, 159.51, 171.87, 143.69, 153.65, 160.25, 153.07, 154.42, 141.21, 154.17, 162.56, 164.44, 172.36, 141.51, 169.4, 167.97, 170.26, 157.8, 167.61]


## 데이터의 평균과 표준편차 구하기

In [15]:
from scipy import stats
import numpy as np

# 평균과 표준편차 계산
mean_value = np.mean(data)
std_dev = np.std(data, ddof=0)  # 모집단 표준편차(기본 설정)

mean_value, std_dev

(np.float64(156.9332), np.float64(9.974187774450611))

## 가설 설정
- 귀무가설: 표본 데이터의 평균은 163과 같다.
- 대립가설: 표본 데이터의 평균은 163과 다르다.

## 정규성 검정
- 수집한 표본 데이터가 정규분포를 따르는지 검증한다.

## 방법 1 : KS-test

In [19]:
# 정규성 검정, KS-test(Kolmogorov-Smirnov) 검정
# KS-test는 주어진 표본 데이터가 특정 이론적인 분포(예: 정규분포)를 따르는지 검정하는 방법

# 비교 대상이 되는 정규분포에 표본 데이터의 평균과 표준편차를 사용
ks_statistic, p_value = stats.kstest(
    data,
    lambda x: stats.norm.cdf(
        x,
        loc=np.mean(data),
        scale=np.std(data, ddof=0)
    )
)

ks_statistic, p_value

(np.float64(0.10925974986314058), np.float64(0.8950958424230889))

In [20]:
stats.shapiro(data) 

ShapiroResult(statistic=np.float64(0.9535807148228921), pvalue=np.float64(0.3014257453674708))